server/src/test/java/org/elasticsearch/index/seqno/RetentionLeaseIT.java

```diff
@@ -397,6 +397,8 @@ public void testRetentionLeasesSyncOnRecovery() throws Exception {
             // check retention leases have been written on the replica; see RecoveryTarget#finalizeRecovery
             assertThat(currentRetentionLeases).isEqualTo(RetentionLeaseUtils.toMapExcludingPeerRecoveryRetentionLeases(replica.loadRetentionLeases()));
         }
+
+        primaryTransportService.clearAllRules();
```

Comment:
Why is this required now?

Reply:
As far as I can tell this should always have been there - otherwise the transport will continue emulating bad behavior during part of the teardown.